src/contracts/core/DelegationManager.sol

```diff
@@ -728,61 +702,84 @@ contract DelegationManager is
         return withdrawalRoot;
     }
 
-    /**
-     * @notice Withdraws `shares` in `strategy` to `withdrawer`. If the shares are virtual beaconChainETH shares, then a call is ultimately forwarded to the
-     * `staker`s EigenPod; otherwise a call is ultimately forwarded to the `strategy` with info on the `token`.
-     */
-    function _withdrawSharesAsTokens(
-        address staker,
-        address withdrawer,
-        IStrategy strategy,
-        uint256 shares,
-        IERC20 token
-    ) internal {
-        if (strategy == beaconChainETHStrategy) {
-            eigenPodManager.withdrawSharesAsTokens({podOwner: staker, destination: withdrawer, shares: shares});
+    /// @dev check whether the withdrawal delay has elapsed (handles both legacy and post-slashing-release withdrawals) and returns the completable timestamp
+    function _checkCompletability(
+        uint32 startTimestamp
+    ) internal view returns (uint32 completableTimestamp) {
+        if (startTimestamp < LEGACY_WITHDRAWALS_TIMESTAMP) {
+            // this is a legacy M2 withdrawal using blocknumbers.
+            // It would take up to 600+ years for the blocknumber to reach the LEGACY_WITHDRAWALS_TIMESTAMP, so this is a safe check.
+            require(startTimestamp + LEGACY_MIN_WITHDRAWAL_DELAY_BLOCKS <= block.number, WithdrawalDelayNotElapsed());
```

Comment:
the real key here is that once the upgrade occurs, no new withdrawals will be stored using block numbers. so as long as the upgrade is completed prior to the block number being `LEGACY_WITHDRAWALS_TIMESTAMP` or higher, this will _never_ be an issue. (perhaps that's the point being made here)